### PR TITLE
Avoid Search with in the same state.

### PR DESCRIPTION
**Pull request by kcherukupalli**
## What’s changing?

To avoid state matching against their own records.  Stripped 

CHANGELOG

## Why?
Relates to Jira Ticket NAC-78

This would speed up the process of matching as we avoid the initiating state.

## This PR has:

- [x] **Commented source code** (_required on public classes and methods, and elsewhere as appropriate_)
- [ ] **Developer documentation** (_for new build/test/API changes or complex portions of the system_)
- [x] **Automated unit tests** (_to maintain or increase level of code coverage_)
- [ ] **Changes to IAC** (_add any deployment steps that require manual intervention to the draft release notes_)

### CODE REVIEW

**Comment by kingcomma** [comment]
@kcherukupalli Are you able to resolve the failing integration tests before I review? Looks like it may be a matter of changing the `InitiatingState` constant to `"eb"` (or something other than `"ea"`)?

**Comment by kcherukupalli** [comment]
@kingcomma  working on integration tests.  Will commit soon.

File: match/docs/openapi/orchestrator/find.yaml [hunk 1]
```diff
@@ -3,7 +3,7 @@ post:
   tags:
     - "Match"
   summary: "Search for all matching participant records using de-identified data"
-  description: "Searches all state databases for any participant records that are an exact match to the `lds_hash` of persons provided in the request."
+  description: "Searches all state(except initiating state) databases for any participant records that are an exact match to the `lds_hash` of persons provided in the request."
```
Comment (kingcomma) [inline match/docs/openapi/orchestrator/find.yaml:6]:
```suggestion
  description: "Searches all state databases for any participant records that are an exact match to the `lds_hash` of persons provided in the request. Does not search the database for the state initiating the request."
```
What do you think about this language? Trying to be as clear as possible. We could also defer to @rserey or @cjhskippy.

*Chore:* Once the language is finalized, can you also update the description in `docs/openapi/duplicate-participation-api.yaml` to match and then re-generate docs. The description for the public-facing markdown documentation is stored there. This should result in a change to `docs/openapi/generate/duplicate-participation-api/openapi.md`.

Comment (kcherukupalli) [reply match/docs/openapi/orchestrator/find.yaml:6]: @kingcomma  looks good to me.

**Comment by kcherukupalli** [comment]
@kingcomma  Added `ReturnsEmptyMatchesFromSameState`  and updated 'Description`

**Review by kcherukupalli — COMMENTED** [review]
@kingcomma  Updated `Description`.

**Review by kingcomma — APPROVED** [review]
👍 👍

**Pull request merged**